### ISSUE TITLE
Process completed with exit code 2

### ISSUE THREAD

**Issue by thommilson** [googlefonts/googlefonts-project-template#109]
After following all of the steps to set up a new repository and getting the initial commit up and running, when new sources are added I get the error code "Process completed with exit code 2." in Build font.

Is there a way I can fix this?

**Comment by jvgaultney** [comment]
Trying to set up a new private project with the template and get the same error on the initial build action (before committing anything):

```
INFO:GFBuilder:Creating variable fonts from Rubik.glyphs
INFO:fontmake.font_project:Building master UFOs and designspace from Glyphs source
INFO:glyphsLib.parser:Parsing .glyphs file
Traceback (most recent call last):
  File "/home/runner/work/font-cadet/font-cadet/venv/bin/gftools", line 8, in <module>
    sys.exit(main())
  File "/home/runner/work/font-cadet/font-cadet/venv/lib/python3.8/site-packages/gftools/scripts/__init__.py", line 91, in main
    mod.main(args[2:])
  File "/home/runner/work/font-cadet/font-cadet/venv/lib/python3.8/site-packages/gftools/builder/__init__.py", line 675, in main
    builder.build()
  File "/home/runner/work/font-cadet/font-cadet/venv/lib/python3.8/site-packages/gftools/builder/__init__.py", line 208, in build
    self.build_variable()
  File "/home/runner/work/font-cadet/font-cadet/venv/lib/python3.8/site-packages/gftools/builder/__init__.py", line 336, in build_variable
    output_files = self.run_fontmake(source, args)
  File "/home/runner/work/font-cadet/font-cadet/venv/lib/python3.8/site-packages/gftools/builder/__init__.py", line 392, in run_fontmake
    FontProject().run_from_glyphs(source, **args)
  File "/home/runner/work/font-cadet/font-cadet/venv/lib/python3.8/site-packages/fontmake/font_project.py", line 904, in run_from_glyphs
    designspace = self.build_master_ufos(
  File "/home/runner/work/font-cadet/font-cadet/venv/lib/python3.8/site-packages/fontTools/misc/loggingTools.py", line 375, in wrapper
    return func(*args, **kwds)
  File "/home/runner/work/font-cadet/font-cadet/venv/lib/python3.8/site-packages/fontmake/font_project.py", line 263, in build_master_ufos
    ufo_path = self._fix_ufo_path(
  File "/home/runner/work/font-cadet/font-cadet/venv/lib/python3.8/site-packages/fontmake/font_project.py", line 175, in _fix_ufo_path
    Path(path).with_suffix(UFO_STRUCTURE_EXTENSIONS[ufo_structure])
KeyError: None
make: *** [Makefile:23: build.stamp] Error 1
Error: Process completed with exit code 2.
```

**Comment by anthrotype** [comment]
> Fontmake 3.7.0 was (from our perspective) a Bad Fontmake.

the problem is you're using fontmake as a library which is not well supported, since fontmake IMO is primarily a CLI tool and I would like to keep that way. I told Marc already once and he acknowledged gftools is doing this at its own risk so to speak..
In fact that code path wasn't tested in fontmake test suite which mostly runs end-to-end from main() and thus the argparse populates the default arguments if not provided on the console.

**Comment by simoncozens** [comment]
That's great. The problem we have is that when users clone the repo, there's a first-run script which calls "pip freeze" and commits that, so that builds are reproducible and subsequent builds don't get modified by changes in the toolchain. So those who cloned during 3.7.0 will get 3.7.0 forever. I guess the better fix here is for users who cloned during 3.7.0's lifetime to change `fontmake==3.7.0` to `fontmake==3.7.1` in their `requirements.txt` and commit that.

If 3.7.0 was only out for a few days, then I suspect Victor's issue was different from the OP's; they just looked the same.

**Comment by anthrotype** [comment]
gftools could bump the minimum required fontmake >= 3.7.1, but still it'd require one to upgrade gftools itself. "Yanking" the 'broken' 3.7.0 release wouldn't fix the problem for those who have frozen their requirements at fontmake==3.7.0.
Isn't there a `make update` in the template's Makefile? That should fix it.

**Comment by simoncozens** [comment]
Yes, make update will fix; I think we're done here since 3.7.0 is no longer current.
